packages/components/inputs/date-input/src/date-input.js

```diff
@@ -125,12 +125,6 @@ const DateInput = (props) => {
           if (changes.hasOwnProperty('highlightedIndex')) {
             setHighlightedIndex(changes.highlightedIndex);
           }
-
-          // ensure calendar always opens on selected item, or on current
-          // month when there is no selected item
-          if (changes.hasOwnProperty('isOpen') && changes.isOpen) {
-            setCalendarDate(props.value === '' ? getToday() : props.value);
-          }
```

Comment:
Removed this block because the `calendarDate` state is already initialized with `props.value || getToday()` so it is not needed.
Furthermore, this was overwriting the state that may have been set in L116, causing inconsistent behaviour.